### PR TITLE
fix skipped nil checks in meshery server handlers

### LINKED ISSUE #7380: Potential panic (skipped nil checks) in meshery server handlers
#### Current Behavior

There are instances in the code where response is not nil checked before accessing properties on it, which might cause panics.
Example- https://github.com/meshery/meshery/blob/70dcb02100709c486ea261337371db5d61676b69/server/models/remote_provider.go#L1881
#### Expected Behavior

Use [ErrUnreachableRemoteProvider](https://github.com/meshery/meshery/blob/master/server/models/error.go#L348) MeshKit error to handle nil response from remote provider.

Checkout this [example](https://github.com/meshery/meshery/blob/master/server/models/remote_provider.go#L3238) use case for reference.
#### Environment

- **Host OS:** Mac Linux Windows
- **Platform:** Docker or Kubernetes
- **Meshery Server Version:** stable-v
- **Meshery Client Version:** stable-v

---
#### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/contributing/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by ayushthe1**
**Notes for Reviewers**

This PR fixes #7380 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Review by leecalcote — CHANGES_REQUESTED** [review]
@Revolyssup will you address these lint issues?

File: server/models/remote_provider.go [hunk 1]
```diff
@@ -518,6 +533,9 @@ func (l *RemoteProvider) SaveK8sContext(token string, k8sContext K8sContext) (K8
 	}
 
 	resp, err := l.DoRequest(cReq, token)
+	if resp == nil {
+		return K8sContext, ErrUnreachableRemoteProvider(err)
```
Comment (Revolyssup) [inline server/models/remote_provider.go:537]: Here err can be nil. Everywhere, first do the err nil check and then resp nil check please.

**Review by Revolyssup — CHANGES_REQUESTED** [review]

File: server/models/remote_provider.go (hunk 1 above)
Comment (ayushthe1) [reply server/models/remote_provider.go]: thanks ,fixed it.

File: server/models/remote_provider.go [hunk 2]
```diff
@@ -111,7 +111,10 @@ func (l *RemoteProvider) loadCapabilities(token string) {
 		// Proceed to make a request with the token
 		resp, err = l.DoRequest(req, token)
 	}
-
+	if resp == nil {
+		logrus.Errorf("Could not reach remote provider %s", err)
```
Comment (theBeginner86) [inline server/models/remote_provider.go:115]:
```suggestion
		logrus.Errorf(ErrUnreachableRemoteProvider(err))
```

File: server/models/remote_provider.go [hunk 3]
```diff
@@ -240,6 +243,10 @@ func (l *RemoteProvider) executePrefSync(tokenString string, sess *Preference) {
 	// }
 	resp, err := l.DoRequest(req, tokenString)
 	if err != nil {
+		if resp == nil {
+			logrus.Errorf("Could not reach remote provider: %s", err)
```
Comment (theBeginner86) [inline server/models/remote_provider.go:247]:
```suggestion
			logrus.Errorf(ErrUnreachableRemoteProvider(err))
```

File: server/models/remote_provider.go (hunk 2 above)
Comment (ayushthe1) [reply server/models/remote_provider.go:115]: Thanks @theBeginner86 for the suggestion

**Comment by leecalcote** [comment]
I think we're still facing some lint issues here.

File: server/models/remote_provider.go [hunk 4]
```diff
@@ -111,7 +111,10 @@ func (l *RemoteProvider) loadCapabilities(token string) {
 		// Proceed to make a request with the token
 		resp, err = l.DoRequest(req, token)
 	}
-
+	if resp == nil {
```
Comment (MUzairS15) [inline server/models/remote_provider.go:114]: Check for error as well, then check resp object

File: server/models/remote_provider.go [hunk 5]
```diff
@@ -111,7 +111,10 @@ func (l *RemoteProvider) loadCapabilities(token string) {
...
+		logrus.Errorf(ErrUnreachableRemoteProvider(err))
```
Comment (MUzairS15) [inline server/models/remote_provider.go:115]:
```suggestion
		logrus.Errorf(ErrUnreachableRemoteProvider(err).Error())
```

Build is failing, please ensure build completes at your local machine before pushing changes.

File: server/models/remote_provider.go [hunk 6]
```diff
@@ -240,6 +243,10 @@ func (l *RemoteProvider) executePrefSync(tokenString string, sess *Preference) {
 	// }
 	resp, err := l.DoRequest(req, tokenString)
 	if err != nil {
+		if resp == nil {
+			logrus.Errorf(ErrUnreachableRemoteProvider(err))
```
Comment (MUzairS15) [inline server/models/remote_provider.go:247]:
```suggestion
			logrus.Errorf(ErrUnreachableRemoteProvider(err).Error())
```

**Review by MUzairS15 — CHANGES_REQUESTED** [review]

**Review by MUzairS15 — APPROVED** [review]

**Comment by Ambareen09** [comment]
Will this PR get merged?

**Review by MUzairS15 — APPROVED** [review]
Thanks @ayushthe1

**Pull request merged**